src/core/positioning/egenioussreceiver.cpp

```diff
@@ -0,0 +1,124 @@
+#include "egenioussreceiver.h"
+
+#include <QHostAddress>
+#include <QJsonDocument>
+#include <QJsonValue>
+
+EgenioussReceiver::EgenioussReceiver( QObject *parent )
+  : AbstractGnssReceiver( parent ), mTcpSocket( new QTcpSocket() )
+{
+  connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
+  connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::handleError );
+  connect( mTcpSocket, &QTcpSocket::connected, this, &EgenioussReceiver::connected );
+  connect( mTcpSocket, &QTcpSocket::disconnected, this, &EgenioussReceiver::disconnected );
+}
+
+void EgenioussReceiver::handleConnectDevice()
+{
+  mTcpSocket->connectToHost( mAddress, mPort, QTcpSocket::ReadWrite );
+}
+
+void EgenioussReceiver::connected()
+{
+  mSocketState = QAbstractSocket::ConnectedState;
+  emit socketStateChanged( mSocketState );
+  setValid( true );
+}
+
+void EgenioussReceiver::handleDisconnectDevice()
+{
+  mTcpSocket->disconnectFromHost();
+}
+
+void EgenioussReceiver::disconnected()
+{
+  if ( mTcpSocket->state() == QAbstractSocket::ConnectedState )
+  {
+    mSocketState = QAbstractSocket::UnconnectedState;
```

Comment:
It feels a bit strange to me that subclasses of postioning device have to manage `mSocketState` that way. mSocketState and mSocketStateString have a quite risk to get out of sync. Wouldn't be it a cleaner interface to reimplement `AbstractGnssReceiver::socketState()` and `socketStateString` in subclasses and get rid of the variables mSocketState and mSocketStateString?

Reply:
@domi4484 , not a bad idea. Can we do that in a follow up PR?